src/transformers/models/gpt2/modeling_gpt2.py

```diff
@@ -715,7 +718,7 @@ def forward(
                 def create_custom_forward(module):
                     def custom_forward(*inputs):
                         # checkpointing only works with tuple returns, not with lists
-                        return tuple(output for output in module(*inputs, use_cache, output_attentions))
```

Comment:
gradient checkpointing was previously only possible with `use_cache=True` which does not make much sense -> change layer accordingly here.